src/boot/efi/meson.build

```diff
@@ -412,43 +412,67 @@ endforeach
 
 foreach tuple : [['systemd-boot@0@.@1@', systemd_boot_objects, false, 'systemd-boot'],
                  ['linux@0@.@1@.stub', stub_objects, true, 'systemd-stub']]
-        elf = custom_target(
-                tuple[0].format(efi_arch[0], 'elf'),
-                input : tuple[1],
-                output : tuple[0].format(efi_arch[0], 'elf'),
-                command : [cc.cmd_array(),
-                           '-o', '@OUTPUT@',
-                           efi_cflags,
-                           efi_ldflags,
-                           '@INPUT@',
-                           '-lefi',
-                           '-lgnuefi',
-                           '-lgcc'],
-                install : tuple[2],
-                install_tag: tuple[3],
-                install_dir : bootlibdir)
-
-        efi = custom_target(
-                tuple[0].format(efi_arch[0], 'efi'),
-                input : elf,
-                output : tuple[0].format(efi_arch[0], 'efi'),
-                command : [objcopy,
-                           '-j', '.bss*',
-                           '-j', '.data',
-                           '-j', '.dynamic',
-                           '-j', '.dynsym',
-                           '-j', '.osrel',
-                           '-j', '.rel*',
-                           '-j', '.sbat',
-                           '-j', '.sdata',
-                           '-j', '.sdmagic',
-                           '-j', '.text',
-                           '--section-alignment=512',
-                           efi_format,
-                           '@INPUT@', '@OUTPUT@'],
-                install : true,
-                install_tag: tuple[3],
-                install_dir : bootlibdir)
+        elf_command = [cc.cmd_array(),
+                   '-o', '@OUTPUT@',
+                   efi_cflags,
+                   efi_ldflags,
+                   '@INPUT@',
+                   '-lefi',
+                   '-lgnuefi',
+                   '-lgcc']
+
+        if meson.version().version_compare('>=0.60.0')
```

Comment:
More install_tag will be added soon - I don't think all this duplication and verbosity is needed, just to save one line in the local CI script. Better to just leave the workaround in the CI script. None of the major distributions use `--fatal-meson-warnings` anyway, it's just in our CI, so it's not very important:

https://git.launchpad.net/ubuntu/+source/systemd/tree/debian/rules
https://tracker.debian.org/media/packages/s/systemd/rules-251rc1-3
https://src.fedoraproject.org/rpms/systemd/blob/rawhide/f/systemd.spec
https://build.opensuse.org/package/view_file/Base:System/systemd/systemd.spec?expand=1

Reply:
> More install_tag will be added soon

I'm not sure why they would be added. If it was discussed somewhere could you point me in the right direction?

> I don't think all this duplication and verbosity is needed

I don't think `install_tags` is needed in its current form either.

> just to save one line in the local CI script

The CI script was just the first place where `systemd` failed to build because `meson` rightfully complained. It's not the only place where it's used.

> None of the major distributions use --fatal-meson-warnings anyway

Is my understand correct that `systemd` doesn't care about backward compatibility in general if major distributions are intact? If so could you add it to `README` to make it a bit more obvious?

Reply:
> Is my understand correct that `systemd` doesn't care about backward compatibility in general if major distributions are intact? If so could you add it to `README` to make it a bit more obvious?

That is not true. We definitely care about backwards compat. But sometimes there's no good answer, and we need to make a choice based on the trade-offs involved. In *that* case, something major like Debian or Fedora weight more significantly, because they simply have the most users, and it's where the developer are, etc. This is particularly true in a case like here, where the issue is not fatal, but more of an annoyance.

Reply:
> But sometimes there's no good answer, and we need to make a choice based on the trade-offs involved

Can't argue with that. I think what I don't understand is why this feature is so important. Was it discussed anywhere apart from https://lists.freedesktop.org/archives/systemd-devel/2022-April/047797.html? By the way even there I haven't been able to find any confirmation that this is actually needed.